### PR TITLE
Remove IDL from Shader subclasses

### PR DESCRIPTION
**Pull request by abarth**
- Gradient
- ImageShader

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @eseidelGoogle

**Comment by abarth** [comment]
I'm going to land this patch TBR @Hixie to unblock.  It's in the same pattern as the earlier patches to remove IDL.

File: sky/engine/core/dart/painting.dart [hunk 1]
```diff
@@ -83,3 +83,76 @@ class ColorFilter extends NativeFieldWrapperClass2 {
...
+  /// Edge is repeated from first color to last.
+  repeated,
+  /// Edge is mirrored from last color to first.
+  mirror
+}
+
+void _validateColorStops(List<Color> colors, List<double> colorStops) {
+  if (colorStops != null && (colors == null || colors.length != colorStops.length)) {
+    throw new ArgumentError(
+        "[colors] and [colorStops] parameters must be equal length.");
```
Comment (Hixie) [inline sky/engine/core/dart/painting.dart]: nit: ) on next line

**Comment by Hixie** [comment]
mostly-rubberstamp-lgtm

**Pull request merged**